apps/aecore/src/aec_peers_pool.erl

```diff
@@ -0,0 +1,1695 @@
+%%
+%% Return a list of peer identifiers and there corresponding extra data given
+%% the **FIRST** time it was added by `update/7`.
+%%
+%% A filtering function can be specified to limit the possible results.
+-spec subset(state(), all | pos_integer(), ext_filter_fun() | undefined)
+  -> {[ext_peer()], state()}.
+subset(St, Size, ExtFilterFun) ->
+  ?ST{rand = RState, use_rand_offset = ROffset, lookup_all = Lookup} = St,
+  IntFilterFun = wrap_filter_fun(St, ExtFilterFun),
+  {PeerIds, RState2} = lookup_sample(Lookup, RState, ROffset, Size, IntFilterFun),
+  Result = export_results(St, PeerIds),
+  {Result, St?ST{rand = RState2}}.
+
+%% @doc Select a random peer from the available ones.
+%%
+%% Takes a time in millisecond as it would be returned from
+%% `erlang:system_time(millisecond)` and uses it as if it was the current time.
+%% Every calls should be given a time greater or equal to the last call.
+%%
+%% The target parameter is used to select from which pool the peer should be
+%% selected:
+%%  - `verified`: Only verified peers will get selected.
+%%  - `unverified`: Only unverified peers will get selected.
+%%  - `both`: A peer will be selected from any of the pools; the pool the peer
+%%    will come from will depend on the configured probability option
+%%    `select_verif_prob`.
+%%
+%% A peer is considered available if it is not selected and if it is not in
+%% standby after being rejected; for a peer to become available again,
+%% `release/3` has to be called or the standby time for rejection has to be
+%% exhausted after calling `reject/3.
+%%
+%% A filter function can be specified to restrict further the set of peer
+%% that can be selected.
+%%
+%% If a peer is available in targeted pools, the call returns the peer
+%% identifier and its extra data given the **FIRST** time it was added by
+%% `update/7`.
+%%
+%% If there is no peers readily available because they are on standby, the call
+%% returns the minimum delay afterwich one will get out of standby.
+%%
+%% When a peer is returned it is considered selected; it means that it will
+%% **NEVER** be returned by `select/3` again until it is released with
+%% `release/2` or marked as rejected with `reject/2` and the standby time is
+%% exausted.
+-spec select(state(), millitimestamp(), select_target(), ext_filter_fun() | undefined)
+  -> {selected, ext_peer(), state()}
+   | {wait, milliseconds(), state()}
+   | {unavailable, state()}.
+select(St, Now, Target, FilterFun) ->
+  select_peer(St, Now, Target, FilterFun).
+
+%% @doc Marks a peer as verified.
+%%
+%% Takes a time in millisecond as it would be returned from
+%% `erlang:system_time(millisecond)` and uses it as if it was the current time.
+%% Every calls should be given a time greater or equal to the last call.
+%%
+%% This call is intended to mark a peer retrieved through other mean than
+%% `select/4` as selected. For example a peer to was just added to the pool
+%% with `update/7` and `verify/3`.
```

Comment:
This paragraph makes no sense 😄

Reply:
Is it better ?
```
%% This call is intended to mark a peer retrieved through other means than
%% @see select/4 as selected. For example, a peer that was just added to the pool
%% with @see update/7 and @see verify/3.
```

Reply:
That's a good explanation. I still think `select/4` vs `selected/3` is confusing.